dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -142,6 +144,52 @@ public User getUser(@ApiParam(hidden = true) @Auth User user) {
         return foundUser;
     }
 
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/user/extended")
+    @ApiOperation(value = "Get additional information on the logged-in user", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = ExtendedUserData.class)
+    public ExtendedUserData getExtendedUserData(@ApiParam(hidden = true) @Auth User user) {
+        User foundUser = userDAO.findById(user.getId());
+        return new ExtendedUserData(foundUser);
+    }
+
+    @POST
+    @Timed
+    @UnitOfWork
+    @Path("/user/changeUsername")
+    @ApiOperation(value = "Change username if possible", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class)
+    public User changeUsername(@ApiParam(hidden = true) @Auth User authUser, @ApiParam("Username to change to") @QueryParam("username") String username) {
+        checkUser(authUser, authUser.getId());
+        Pattern pattern = Pattern.compile("^[a-zA-Z]+[.a-zA-Z0-9-_]*$");
```

Comment:
You could make the Pattern a static so that it's only compiled once, rather than on every invocation. Not that it will get invoked a lot, and will probably have no noticeable performance difference...